sphinxcontrib/jsonschema.py

```diff
@@ -87,11 +105,18 @@ def run(self):
                                        self.arguments[0])
                 env.note_dependency(relpath)
 
-                schema = JSONSchema.loadfromfile(abspath, allow_external_refs=('allowexternalrefs' in self.options))
+                schema = JSONSchema.loadfromfile(
+                    abspath,
+                    allow_external_refs=('allowexternalrefs' in self.options),
+                    loader=(partial(urnjsonloader, abspath.rsplit("/", 1)[0])
+                            if 'allowurnrefs' in self.options else None)
+                )
             else:
                 schema = JSONSchema.loadfromfile(
                     ''.join(self.content),
-                    allow_external_refs=('allowexternalrefs' in self.options)
+                    allow_external_refs=('allowexternalrefs' in self.options),
+                    loader=(partial(urnjsonloader, abspath.rsplit("/", 1)[0])
+                            if 'allowurnrefs' in self.options else None)
```

Comment:
could do 
```python
if options.get("allowurnrefs")
```
?

Reply:
You might think that should work, but you would be over estimating the design of the Sphinx directives, if a flag is present you get something like `options = {'allowurnrefs': None}`, and so `bool('allowurnrefs' in options) -> True` but `bool(options.get('allowurnrefs')) -> False`. If it resulted in `options = {'allowurnrefs': True}`, like it probably should, then this would work.